Mazebuildertest

halephil opened the pull request:
AddLights method
-picks every other cells to add spot lights to
AddLight 
-Adds light to cells that Add Lights picks
Fixed method for ChangeXYZPos

Commit:
-Added method MazeName(Set;Get;)
-Added method SetXYZMazeScale

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into MazeBuilderTest-

Commit:
SetXYZPosition, SetXYZScale for Maze Class, Draw Maze Changes

SetXYZPosition, SetXYZScale for Maze Class
-fixed both methods so they properly adjust scale and position of Maze
MazeDrawer Changes
-omitted gameobject variable in class
-children and parent GameObjects are now added to the Maze object being
passed into the MazeDrawer Class

Commit:
Light Method

Adds lights over each cell

Commit:
Merged Files To accept both changes in Alex's revision and My revision

Commit:
Position Method

Changes position of Maze

Commit:
UpdateVariaablesName

after merge a few errors occured because of previous commints I made

Commit:
AddLight/AddLights

AddLights
-Locates which cells to add a light to
AddLight
-Adds a light with the correct properties to the cells selected by add
light

alexweav commented on VR Maze/Assets/Scripts/Maze.cs:
```diff
@@ -1,81 +1,125 @@
+        /// <param name="southPath"> bool value if a south path exist to another cell </param>
+        public void addMazeCell(int x, int z, bool eastPath, bool southPath)
+        {
+            MazeCell currentCell = new MazeCell(x, z, southPath, eastPath);  //Creates new mazeCell
```
This is a bug, southPath and eastPath are getting reversed here. The MazeCell constructor takes these arguments in reverse order (eastPath first). This causes one of the tests to fail, I fixed it in my last merge but it's reverted here

alexweav commented on VR Maze/Assets/Scripts/Maze.cs:
```diff
@@ -1,81 +1,125 @@
+            mazeName = MazeName;
+            ThisMaze = new GameObject(mazeName);
+            ThisMazeScale = new Vector3(.5f, 10f, .5f);
+            ThisMazePosition = new Vector3(0f, 0f, 0f);
```
Should this stuff be added to the other constructor as well? If anyone uses the other constructor then all this stuff will be undefined

alexweav commented on VR Maze/Assets/Scripts/MazeDrawer.cs:
```diff
@@ -9,24 +9,22 @@ public class MazeDrawer : MazeGenerate
         {
-            mazeCellList = MazeToDraw.CellsInMaze;
-            totalMaze = new GameObject("Maze");
+            mazeCellList = (List<MazeCell>)MazeToDraw.cellsInMaze;
```
I fixed the need for this typecast in my last commit but it's undone here

alexweav reviewed (changes requested):
The CI should have not passed since this breaks tests, i'll look into it

Commit:
Added Intialize Maze method, Fixed 2 Merge Bugs

Intialize
-Sets the parameters for a new maze such as size, and position
Fixed Merge Bugs
-Fixed Type cast issue in MazeDrawer.cs
-Fixed switched parameters in Maze.cs  addMazeCell method

Commit:
Requsted Changes

-Last Commit did not include the Merge bug fixes
this commit does
